MHRA/products
# Delete blob before deleting entry from search index. Addresses #653

Delete blob before deleting entry from search index.

Addresses #653.

![](https://media.giphy.com/media/R9eXzDRi7Abcs/giphy.gif)

### Acceptance Criteria

- [ ] If deleting from blob storage fails, the delete job should be able to be retried.

### Testing information

To test, I:

* put a document into the system;
* changed the environment variable `STORAGE_CONTAINER` to a garbage value
* restarted the doc index updater
* sent a request to delete the document
* saw that the delete failed
* saw that the delete was retried
* changed `STORAGE_CONTAINER` back to its original value
* restarted the doc index updater
* saw that the delete was retried and succeeded

Note that the job status is updated after every attempt to process the delete message, this should be fixed in #658
